tests/benchmark/utils/s2n_server_recv_benchmark.cc

```diff
@@ -0,0 +1,213 @@
+
+    strcat(bench_out, file_prefix.c_str());
+    std::string log_output_name = "server_" + file_prefix;
+    FILE* write_log = freopen(log_output_name.c_str(), "w", stdout);
+
+    std::vector<char*> argv_bench(argv, argv + argc);
+    argv_bench.push_back(bench_format);
+    argv_bench.push_back(bench_out);
+    argv_bench.push_back(nullptr);
+    argc = argv_bench.size();
+
+    const char *session_ticket_key_file_path = NULL;
+
+    int setsockopt_value = 1;
+    int max_early_data = 0;
+    conn_settings.session_ticket = 1;
+    conn_settings.session_cache = 0;
+    conn_settings.max_conns = -1;
+    conn_settings.psk_list_len = 0;
+    conn_settings.insecure = insecure;
+    conn_settings.use_corked_io = use_corked_io;
+
+    s2n_init();
+
+    hints.ai_family = AF_UNSPEC;
+    hints.ai_socktype = SOCK_STREAM;
+
+    if (signal(SIGPIPE, SIG_IGN) == SIG_ERR) {
+        fprintf(stderr, "Error disabling SIGPIPE\n");
+        exit(1);
+    }
+
+    GUARD_EXIT(getaddrinfo(host, port, &hints, &ai), "getaddrinfo error\n");
+    GUARD_EXIT((sockfd = socket(ai->ai_family, ai->ai_socktype, ai->ai_protocol)), "socket error\n");
+    GUARD_EXIT(setsockopt(sockfd, SOL_SOCKET, SO_REUSEADDR, &setsockopt_value, sizeof(int)), "setsockopt error\n");
+    GUARD_EXIT(bind(sockfd, ai->ai_addr, ai->ai_addrlen), "Server bind error\n");
+    GUARD_EXIT(listen(sockfd, 1), "listen error\n");
+
+    if (DEBUG_PRINT) {
+        printf("Listening on %s:%s\n", host, port);
+    }
+
+    config = s2n_config_new();
+    if (!config) {
+        print_s2n_error("Error getting new s2n config");
+        exit(1);
+    }
+
+    s2n_set_common_server_config(max_early_data, config, conn_settings, cipher_prefs, session_ticket_key_file_path);
+
+    struct s2n_cert_chain_and_key *chain_and_key_rsa = s2n_cert_chain_and_key_new();
+    GUARD_EXIT(s2n_cert_chain_and_key_load_pem(chain_and_key_rsa, rsa_certificate_chain, rsa_private_key),
+               "Error loading RSA certificate/key");
+
+    GUARD_EXIT(s2n_config_add_cert_chain_and_key_to_store(config, chain_and_key_rsa),
+               "Error adding RSA chain and key");
+
+    struct s2n_cert_chain_and_key *chain_and_key_ecdsa = s2n_cert_chain_and_key_new();
+    GUARD_EXIT(s2n_cert_chain_and_key_load_pem(chain_and_key_ecdsa, ecdsa_certificate_chain, ecdsa_private_key),
+               "Error loading ECDSA certificate/key");
+
+    GUARD_EXIT(s2n_config_add_cert_chain_and_key_to_store(config, chain_and_key_ecdsa),
+               "Error adding ECDSA chain and key");
+
+    bool stop_listen = false;
+    while ((!stop_listen) && (connectionfd = accept(sockfd, ai->ai_addr, &ai->ai_addrlen)) > 0) {
+        for (unsigned int suite_num = 0; suite_num < num_suites; ++suite_num) {
+            std::string bench_name = std::string("Server: ") + all_suites[suite_num]->name;
+
+            int kb_1 = 1000;
+            int kb_10 = 10000;
+            int mb_1 = 1000000;
```

Comment:
Can this get pulled into an array/optional parameter so users can change the size more easily. Also I'm interested in some more sizes by default: 1 byte, 100 bytes, 1 kb, 10kb, 100 kb, 1 mb.